src/converter/modules/ExlClient.js

```diff
@@ -93,6 +95,53 @@ export default class ExlClient {
     throw new Error(`Playlist with id: ${id} not found`);
   }
 
+  /**
+   * Get non-English labels from EXL Config populated endpoints
+   * @param {string} endpoint
+   * @param {string} id
+   * @param {string} lang
+   * @returns {string}
+   */
+  async getLabelFromEndpoint(endpoint, id, lang = 'en') {
+    if (LABELS_FROM_ENDPOINTS[endpoint] === undefined) {
+      LABELS_FROM_ENDPOINTS[endpoint] = {};
+      LABELS_FROM_ENDPOINTS[endpoint][lang] = {};
+    }
+
+    if (LABELS_FROM_ENDPOINTS.endpoint?.lang === undefined) {
+      let next = `api/${endpoint}?lang=${lang}&page_size=2000`;
+
+      do {
+        /* eslint-disable-next-line no-await-in-loop */
+        const response = await this.doFetch(next);
+
+        if (response.error) {
+          console.error(response.error);
+        } else {
+          const raw = response?.data;
+
+          if (raw === undefined || raw.length <= 0) {
+            console.error(`${endpoint} request returned no labels for ${lang}`);
+          }
+
+          raw.forEach((item) => {
+            const enLabel =
+              item.Name_en === undefined ? item?.Name : item?.Name_en;
+            LABELS_FROM_ENDPOINTS[endpoint][lang][enLabel] = item.Name;
+          });
+        }
+
+        // "Next" is always used when page size < remaining items, "Last" is used when page size > items remaining
+        const nextUriObject =
+          response?.links.find((link) => link.rel === 'next') ||
+          response?.links.find((link) => link.rel === 'last');
+        next = nextUriObject?.uri;
+      } while (next !== undefined);
+    }
```

Comment:
Seems unnecessary to handle a scenario we would never run into, no?

Reply:
I ran into it multiple times while testing. API puts arbitrary limits on some API responses, so even if you request 2000, you might only get 200. It also covers the case where features keep getting added in excess of whatever page size we request. Not using a page size uses whatever the default page size is for the API, so there isn't a way atm to just request all values without pagination atm.